### PR TITLE
fix MOSDOp encoding

### PR DESCRIPTION
**Pull request by liewegas**
We were previously using clenit_id in print() even though it was not decoded,
and the reqid handling was semi-broken.

### CODE REVIEW

**Comment by tchaikov** [comment]
lgtm with a nit which might not make any sense.

**Comment by liewegas** [comment]
@tchaikov I cleaned this up, and fixed the v6 case too (which was totally broken.. reqid is at the very end!).

This'll need to go through some upgrade tests before merging, but see if it looks right first?

File: src/messages/MOSDOp.h [hunk 1]
```diff
@@ -116,16 +153,25 @@ class MOSDOp : public Message {
...
       osdmap_epoch(_osdmap_epoch), flags(_flags), retry_attempt(-1),
-      oid(_oid), oloc(_oloc), pgid(_pgid), partial_decode_needed(false), final_decode_needed(false),
+      oid(_oid), oloc(_oloc), pgid(_pgid),
+      partial_decode_needed(false),
+      final_decode_needed(false),
       features(feat) {
     set_tid(tid);
+
+    // also put the client_inc in reqid.inc, so that get_reqid() can
+    // be used before the full message is decode.
```
Comment (tchaikov) [inline src/messages/MOSDOp.h]: s/decode/decoded/

**Comment by tchaikov** [comment]
lgtm other than a typo and the order of checking `final_decode_needed` and `header.version` in `finish_decode()`.

**Comment by liewegas** [comment]
fixed those 2 things, thanks!

**Comment by tchaikov** [comment]
lgtm after the qa run passes.

**Pull request merged**